### ISSUE TITLE
C++ API examples

### ISSUE THREAD

**Issue by okpatil4u** [ml-explore/mlx-examples#128]
Could you please add a few examples on how one could consume c++ API ? Especially how one could use them with Rust ?

Our entire production system is based on rust. We would love if you could give us any pointers on this.

**Comment by awni** [comment]
Yes there are a few examples in [the core repo](https://github.com/ml-explore/mlx/tree/main/examples/cpp). We don't have a higher level `nn` API in C++ yet. That's on the TODO list.

Is that the kind of thing you are looking for? If not could you be more specific?

**Comment by okpatil4u** [comment]
A simple example of running 16/8bit llama inference through c++ api should
be sufficient. I can then review how I can use it through rust.
